discovery-provider/src/challenges/challenge_event_bus.py

```diff
@@ -29,34 +27,44 @@ def register_listener(self, event, listener):
 
     def dispatch(self, session, event, block_number, user_id):
         """Dispatches an event + block_number + user_id to Redis queue"""
-        event_json = self._event_to_json(event, block_number, user_id)
-        logger.debug(f"ChallengeEventBus: dispatch {event_json}")
-        self._redis.rpush(REDIS_QUEUE_PREFIX, event_json)
+        try:
```

Comment:
i don't think you need the session in dispatch here (and it's great that we don't b/c this doesn't block indexing at all)